hibernate-types-4/src/main/java/com/vladmihalcea/hibernate/type/range/guava/PostgreSQLGuavaRangeType.java

```diff
@@ -121,14 +127,18 @@ private static String determineRangeType(Range<?> range) {
             upper = converter.apply(upperStr);
         }
 
+        if (lower == null && upper == null) {
+            throw new IllegalArgumentException("Cannot find bound type");
+        }
+
         if (lowerStr.length() == 0) {
             return upperBound == BoundType.CLOSED ?
                 Ranges.atMost(upper) :
                 Ranges.lessThan(upper);
         } else if (upperStr.length() == 0) {
             return lowerBound == BoundType.CLOSED ?
-                Ranges.atLeast(upper) :
-                Ranges.greaterThan(upper);
+                Ranges.atLeast(lower) :
```

Comment:
This fix is actually necessary for the proper working of the type, I have it applied locally as well (just hadn't the time to push it upstream yet).

Not sure about the other change yet. We don't "fix" the support but just throw another type of error. I think we can do more.